### PR TITLE
CIV-869 Email notification to parties SDO

### PR DESCRIPTION
**Pull request by Gulam-Hussain**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/CIV-869

### Change description ###
This PR creates a handler for the `CREATE_SDO` journey as well as handlers for the two notification events created in [this PR](https://github.com/hmcts/civil-ccd-definition/pull/373/files). Along side these class files are the tests.

The `CreateSDOCallbackHandler` was created based on the `CreateClaimCallbackHandler` taking out everything that was not needed at this time. 

The message `Header` and `Body` used are just placeholders until we decide upon what to display to the user, this is the same for the email templates used in the notification handlers and will be changed in a future PR.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/CreateSDOApplicantNotificationHandler.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+import static uk.gov.hmcts.reform.civil.callback.CallbackType.ABOUT_TO_SUBMIT;
+import static uk.gov.hmcts.reform.civil.callback.CaseEvent.NOTIFY_APPLICANT_SOLICITOR1_SDO_TRIGGERED;
+
+@Service
+@RequiredArgsConstructor
+public class CreateSDOApplicantNotificationHandler extends CallbackHandler implements NotificationData {
+
+    private static final List<CaseEvent> EVENTS = List.of(NOTIFY_APPLICANT_SOLICITOR1_SDO_TRIGGERED);
+
+    private static final String REFERENCE_TEMPLATE = "case-taken-offline-applicant-notification-%s";
```
Comment (Gulam-Hussain) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/CreateSDOApplicantNotificationHandler.java:27]: This is temporary and will be updated in a later PR to make it relevant to SDO

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/CreateSDOApplicantNotificationHandler.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    public List<CaseEvent> handledEvents() {
+        return EVENTS;
+    }
+
+    private CallbackResponse notifyApplicantSolicitorSDOTriggered(CallbackParams callbackParams) {
+        CaseData caseData = callbackParams.getCaseData();
+
+        notificationService.sendMail(
+            caseData.getApplicantSolicitor1UserDetails().getEmail(),
+            notificationsProperties.getSolicitorCaseTakenOffline(),
```
Comment (Gulam-Hussain) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/CreateSDOApplicantNotificationHandler.java:55]: This template is temporary and will be updated in a later PR to make it relevant to SDO

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/CreateSDORespondentNotificationHandler.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+import static uk.gov.hmcts.reform.civil.callback.CallbackType.ABOUT_TO_SUBMIT;
+import static uk.gov.hmcts.reform.civil.callback.CaseEvent.NOTIFY_RESPONDENT_SOLICITOR1_SDO_TRIGGERED;
+
+@Service
+@RequiredArgsConstructor
+public class CreateSDORespondentNotificationHandler extends CallbackHandler implements NotificationData {
+
+    private static final List<CaseEvent> EVENTS = List.of(NOTIFY_RESPONDENT_SOLICITOR1_SDO_TRIGGERED);
+
+    private static final String REFERENCE_TEMPLATE = "case-taken-offline-respondent-notification-%s";
```
Comment (Gulam-Hussain) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/CreateSDORespondentNotificationHandler.java:27]: This is temporary and will be updated in a later PR to make it relevant to SDO

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/CreateSDORespondentNotificationHandler.java [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    public List<CaseEvent> handledEvents() {
+        return EVENTS;
+    }
+
+    private CallbackResponse notifyRespondentSolicitorSDOTriggered(CallbackParams callbackParams) {
+        CaseData caseData = callbackParams.getCaseData();
+
+        notificationService.sendMail(
+            caseData.getRespondentSolicitor1EmailAddress(),
+            notificationsProperties.getSolicitorCaseTakenOffline(),
```
Comment (Gulam-Hussain) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/CreateSDORespondentNotificationHandler.java:55]: This template is temporary and will be updated in a later PR to make it relevant to SDO

File: src/test/java/uk/gov/hmcts/reform/civil/handler/callback/user/CreateSDOCallbackHandlerTest.java [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+    public static final String REFERENCE_NUMBER = "000DC001";
+
+    @MockBean
+    private Time time;
+
+    @MockBean
+    private IdamClient idamClient;
+
+    @MockBean
+    private FeatureToggleService featureToggleService;
```
Comment (johnoneill46) [inline src/test/java/uk/gov/hmcts/reform/civil/handler/callback/user/CreateSDOCallbackHandlerTest.java:65]: This necessary if you're not using any toggles?

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/CreateSDOCallbackHandler.java [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+import static uk.gov.hmcts.reform.civil.helpers.DateFormatHelper.DATE_TIME_AT;
+import static uk.gov.hmcts.reform.civil.helpers.DateFormatHelper.formatLocalDateTime;
+
+@Service
+@RequiredArgsConstructor
+public class CreateSDOCallbackHandler extends CallbackHandler {
+
+    private static final List<CaseEvent> EVENTS = Collections.singletonList(CREATE_SDO);
+
+    public static final String CONFIRMATION_SUMMARY = "<br/>[Download the sealed claim form](%s)"
```
Comment (Gulam-Hussain) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/CreateSDOCallbackHandler.java:39]: This is temporary and will be updated in a later PR to make it relevant to SDO

**Review by johnoneill46 — APPROVED** [review]

File: src/test/java/uk/gov/hmcts/reform/civil/handler/callback/user/CreateSDOCallbackHandlerTest.java (hunk 5 above)
Comment (Gulam-Hussain) [reply src/test/java/uk/gov/hmcts/reform/civil/handler/callback/user/CreateSDOCallbackHandlerTest.java]: Good catch, must've forgot to remove that

**Comment by Gulam-Hussain** [comment]
Closed as changes are included within the following PR that updates the email template used: https://github.com/hmcts/civil-service/pull/664

**Pull request closed without merging**